packages/astro/src/runtime/server/render/util.ts

```diff
@@ -28,6 +28,7 @@ const kebab = (k: string) =>
 	k.toLowerCase() === k ? k : k.replace(/[A-Z]/g, (match) => `-${match.toLowerCase()}`);
 const toStyleString = (obj: Record<string, any>) =>
 	Object.entries(obj)
+		.filter(([k, v]) => v !== undefined && v !== null)
```

Comment:
Can't we use `.filter(Boolean)`?

Reply:
https://github.com/withastro/astro/pull/8940#discussion_r1376193061 See my test. `font-size: 0` is a valid value